src/back/controllers/api/routes/categories/categories.js

```diff
@@ -1,19 +1,44 @@
 'use strict';
 
-const category = require('./../../models/category');
-const user = require('./../../models/user');
+const categoryModel = require('./../../models/category');
+const userModel = require('./../../models/user');
 
 const getAll = (req, res) => {
-    category.find({}, (err, categories) => {
-        res.json(categories);
+    categoryModel.find({}, (err, findedCategories) => {
+        if (err) res.sendStatus(500);
+
+        const userId = req.user._id;
+        let categories;
+
+        userModel.findById(userId, (error, user) => {
+            if (error) res.sendStatus(500);
+
+            if (user.categories) {
+                categories = findedCategories.map((category) => {
```

Comment:
what does this anonymous function states for? it is not so simple do not force to read implementation, to understand it